### PR TITLE
contract selection init

### PR DESCRIPTION
**Pull request by Aniket6990**

### CODE REVIEW

File: src/utils/functions.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import * as fs from "fs";
+import path from "path";
+import { window, workspace } from "vscode";
+
+export const getSourceName = (contractTitle: string) => {
+  if (workspace.workspaceFolders === undefined) {
+    window.showErrorMessage("please open solidity project to work");
```
Comment (Krish-bhardwaj) [inline src/utils/functions.ts:7]: This condition is not specifically checking if a solidity projects is opened or not we can modify this as `open a project` or add proper check condition that checks if a `hardhat project` is open or not

File: src/utils/functions.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    );
+
+    const sourceName: string = filePathArray[0].sourceName;
+
+    return fs.existsSync(
+      path.join(
+        path_,
+        "artifacts",
+        sourceName,
+        `${contractTitle}_deployed_address.json`
```
Comment (Krish-bhardwaj) [inline src/utils/functions.ts:33]: We can read contract addressed from the `ethcode API ` we dont need to use  node.js `fs` module . If there are any more requirements on the API we can develop that .

**Review by Krish-bhardwaj — CHANGES_REQUESTED** [review]

File: src/utils/functions.ts (hunk 2 above)
Comment (Aniket6990) [reply src/utils/functions.ts:33]: Here we are checking `fileName_deployed_address.json` file for the contract that we are selecting is present or not.

Comment (Krish-bhardwaj) [reply src/utils/functions.ts:33]: we have an api for reading the contract address probably we can add use that if no data is retuned that means no address is available and if the file is not present the contract would not show  in the list contract api

Comment (Krish-bhardwaj) [reply src/utils/functions.ts:33]: image

**Pull request merged**